soc/ti/k3/am6x/m4/soc.c

```diff
@@ -71,3 +72,16 @@ static int am62x_m4_init(void)
 }
 
 SYS_INIT(am62x_m4_init, EARLY, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+#endif
+
+#if CONFIG_SOC_AM6442_M4
+static int am64x_m4_init(void)
+{
+	sys_mm_drv_ti_rat_init(
+		region_config, ADDR_TRANSLATE_RAT_BASE_ADDR, ARRAY_SIZE(region_config));
+	mmr_unlock();
```

Comment:
What was the intent to duplicate this?  I see you are using the exact same RAT configs for AM62 and AM64, I don't think that is a good idea even though basic functionality works fine, there should be some flexibility to configure this per SoC with different memory map as one mapping configuration might not work everywhere.

Reply:
I'm not totally sure both architectures are 100% identical.

I thought about renaming the am62 init to `am6x_m4_init`. Guess I will do that...

Reply:
@dnltz sorry I think my comment was confusing, what I meant was that the intent to duplicate should be also to use separate RAT configurations, having the same is not a good idea for long term when more and more support is added.